File: qlty-cli/tests/lang/cpp/basic.in/NestedControl.cpp
#include <iostream>
#include <vector>
#include <stdexcept>

class NestedControl {
public:
    static void notNested(const std::string& foo, const std::string& bar) {
        if ((foo == "cat" && bar == "dog") || (foo == "dog" && bar == "cat")) {
            std::cout << "Got a cat and a dog!" << std::endl;
        } else {
            std::cout << "Got nothing" << std::endl;
        }
    }

    static void deeplyNested(bool a, bool b, bool c, bool d, bool e, bool f) {
        if (a) {
            for (int i = 0; i < 10; i++) {
                if (b) {
                    try {
                        if (c) {
                            for (auto& val : std::vector<int>{1, 2, 3}) {
                                if (d) {
                                    std::cout << "deep: " << val << std::endl;
                                }
                            }
                        }
                    } catch (const std::exception& ex) {
                        std::cout << "error" << std::endl;
                    }
                }
            }
        }
    }
};

File: qlty-cli/tests/lang/vbnet/basic.in/NestedControl.vb
Imports System

Public Class Main
    Public Shared Sub NotNested(foo As String, bar As String)
        If (foo = "cat" AndAlso bar = "dog") OrElse (foo = "dog" AndAlso bar = "cat") Then
            Console.WriteLine("Got a cat and a dog!")
        Else
            Console.WriteLine("Got nothing")
        End If
    End Sub

    Public Shared Sub F0(bar As Boolean, baz As Boolean, qux As Boolean, quux As Boolean)
        If bar Then
            If baz Then
                If qux Then
                    If quux Then
                        Console.WriteLine("Not deeply nested enough!")
                    End If
                End If
            End If
        End If
    End Sub

    Public Shared Function F2(foo As Integer) As String
        Select Case foo
            Case 1
                Return "bar1"
            Case 2
                Return "bar2"
            Case 3
                Return "bar3"
            Case 4
                Return "bar4"
            Case 5
                Return "bar5"
            Case 6
                Return "bar6"
            Case 7
                Return "bar7"
            Case 8
                Return "bar8"
            Case 9
                Return "bar9"
            Case 10
                Return "bar10"
            Case Else
                Throw New ArgumentException("Invalid foo value")
        End Select
    End Function
End Class
